taichi/backends/cc/codegen_cc.cpp

```diff
@@ -533,6 +533,9 @@ class CCTransformer : public IRVisitor {
 
   void visit(AdStackAllocaStmt *stmt) override {
     TI_ASSERT(stmt->width() == 1);
+    TI_ASSERT_INFO(
+        stmt->max_size > 0,
+        "Adaptive autodiff stack's size should have been determined.");
```

Comment:
Since this string appears in all backends, maybe we should make it a global constant. @k-ye WDYT? Should we create something link `error_messages.h` to keep the constant string literals?

Reply:
Some thoughts here: we may want to add a pass like `verify_before_codegen()`. Some assertions such as this one and "global ptrs should have been lowered" are common across all backends.

Reply:
That makes sense to me!